pkgs/applications/science/biology/neuron/default.nix

```diff
@@ -1,94 +1,104 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchurl
-, pkg-config
-, automake
-, autoconf
-, libtool
-, ncurses
 , readline
-, which
-, python ? null
-, useMpi ? false
+, xorg
 , mpi
-, iv
+, cmake
+, bison
+, flex
+, git
+, perl
+, gsl
+, xcbuild
+, python3
+, useMpi ? false
+, useIv ? true
+, useCore ? false
+, useRx3d ? false
 }:
 
+
 stdenv.mkDerivation rec {
-  pname = "neuron${lib.optionalString useMpi "-mpi"}";
-  version = "7.5";
+  pname = "neuron";
+  version = "8.2.1";
 
-  nativeBuildInputs = [ which pkg-config automake autoconf libtool ];
-  buildInputs = [ ncurses readline python iv ]
-    ++ lib.optional useMpi mpi;
+  # format is for pythonModule conversion
+  format = "other";
 
-  src = fetchurl {
-    url = "https://www.neuron.yale.edu/ftp/neuron/versions/v${version}/nrn-${version}.tar.gz";
-    sha256 = "0f26v3qvzblcdjg7isq0m9j2q8q7x3vhmkfllv8lsr3gyj44lljf";
-  };
+  nativeBuildInputs = [
+    cmake
+    bison
+    flex
+    git
+  ] ++ lib.optional useCore [ perl gsl ]
+  ++ lib.optional stdenv.isDarwin [ xcbuild ];
 
-  patches = (lib.optionals (stdenv.isDarwin) [ ./neuron-carbon-disable.patch ]);
+  buildInputs = lib.optional useIv [
+    xorg.libX11
```

Comment:
apparently it wants x for darwin

```
In file included from /tmp/nix-build-neuron-8.2.1.drv-0/nrn/external/iv/src/lib/IV-2_6/xbitmap2_6.cpp:37:
In file included from /tmp/nix-build-neuron-8.2.1.drv-0/nrn/external/iv/src/include/IV-X11/xbitmap.h:29:
/tmp/nix-build-neuron-8.2.1.drv-0/nrn/external/iv/src/include/IV-X11/Xlib.h:57:10: fatal error: 'X11/Xlib.h' file not found
#include <X11/Xlib.h>
         ^~~~~~~~~~~~
1 error generated.
make[2]: *** [external/iv/src/lib/CMakeFiles/interviews.dir/build.make:696: external/iv/src/lib/CMakeFiles/interviews.dir/IV-2_6/xbitmap2_6.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:305: external/iv/src/lib/CMakeFiles/interviews.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/xp0wlxg98zfw00ji2qfa23c5437nmg64-neuron-8.2.1.drv' failed with exit code 2;
```

Reply:
Curious that that comes in and out of being needed, I have since fixed this issue @SuperSandro2000 with:

`xorg.libX11.dev` instead of `xorg.libX11`. Built on my Mac and Linux machine, hopefully borg confirms